pkg/services/auth/jwt/auth.go

```diff
@@ -42,13 +47,46 @@ func (s *AuthService) Init() error {
 	if err := s.initClaimExpectations(); err != nil {
 		return err
 	}
+	if s.Cfg.JWTAuthKeyFile == "" && s.Cfg.JWTAuthJWKSetFile == "" && s.Cfg.JWTAuthJWKSetURL == "" {
+		if err := s.detectJWKSUri(); err != nil {
+			s.log.Debug("JWKS uri not detected due to", "err", err)
+		}
+	}
 	if err := s.initKeySet(); err != nil {
 		return err
 	}
 
 	return nil
 }
 
+// Auto configure JWKS Url following OIDC standards if the issuer is trusted
+// This can't work if you don't trust any issuers in JWTAuthExpectClaims
+func (s *AuthService) detectJWKSUri() error {
+	if s.expectRegistered.Issuer != "" {
+		oidcConfigURL := strings.TrimSuffix(s.expect["iss"].(string), "/") + OidcWellKnowConfigUri
+		var httpClient = &http.Client{Timeout: 10 * time.Second}
```

Comment:
Probably want to replace this with HTTP client provider, but we can discuss that later.